test/test_treetensornetworks/test_solvers/test_contract.jl

```diff
@@ -88,4 +88,21 @@ end
   @test inner(psit, Hpsi) ≈ inner(psit, H, psi) atol = 1E-4
 end
 
+@testset "Contract TTN with dangling inds" begin
+  nbit = 3
+  sites = siteinds("Qubit", nbit)
+
+  # randomMPO does not support linkdims keyword.
+  M1 = replaceprime(randomMPO(sites) + randomMPO(sites), 1=>2, 0=>1)
+  M2 = randomMPO(sites) + randomMPO(sites)
+  M12_ref = contract(M1, M2; alg="naive")
+  t12_ref = TreeTensorNetwork([M12_ref[v] for v in eachindex(M12_ref)])
+
+  t1 = TreeTensorNetwork([M1[v] for v in eachindex(M1)])
+  t2 = TreeTensorNetwork([M2[v] for v in eachindex(M2)])
+
+  # Test with good initial guess
+  @test contract(t1, t2; alg="fit", init=t12_ref) ≈ t12_ref
```

Comment:
You should a tolerance here since by default the comparison is very strict, maybe `rtol = 1e-7`.